tests/test_readwrite.py

```diff
@@ -344,13 +345,14 @@ def test_zarr_compression(tmp_path):
 
     ad.io.write_zarr(pth, adata, compressor=compressor)
 
-    def check_compressed(key, value):
-        if isinstance(value, zarr.Array) and value.shape != ():
+    def check_compressed(value, key):
+        if value.shape != ():
             if value.compressor != compressor:
                 not_compressed.append(key)
 
-    with zarr.open(str(pth), "r") as f:
-        f.visititems(check_compressed)
+    f = zarr.open(str(pth), mode="r")
+    for key in f.array_keys():
+        check_compressed(f[key], key)
```

Comment:
doesn’t visititems go all the way down? Is that not necessary for this test?

Reply:
`visititems` doesn't exist on zarr v3 :/

Reply:
My question was if the structure in the test can be deeper than one level. If so, the code you replaced `visititems` with doesn’t do the same thing.

Reply:
apologies, shouldn't have assumed that it would go deeper!